### PR TITLE
Fix Image reflection issue and proguard rules

### PR DESCRIPTION
**Pull request by ambushwork**
### What does this PR do?

* Changes some proguard rules in order to make the code work in release build.
* Changes some reflection logic for image mapper in order to make it work in higher version of `Coil`
* Change the bitmap copy config in order to make it work for Bitmap ALPHA config.
* Update the sample app image showcase screen.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: features/dd-sdk-android-session-replay-compose/consumer-rules.pro [hunk 1]
```diff
@@ -43,10 +43,16 @@
...
 }
--keepclassmembers class coil.compose.ContentPainterModifier {
+-keep class coil.compose.ContentPainterModifier {
      <fields>;
 }
--keepclassmembers class coil.compose.AsyncImagePainter {
+-keep class coil.compose.AsyncImagePainter {
      <fields>;
 }
+-keepclassmembers class androidx.compose.foundation.layout.PaddingElement{
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/consumer-rules.pro:52]: in the changes above we are _sometimes_ replacing usages of `keepclassmembers` to just `keep`, so is using it here won't cause any issues?

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt [hunk 2]
```diff
@@ -82,7 +86,7 @@ internal class ImageSemanticsNodeMapper(
 
         val newBitmap = bitmap?.let {
             @Suppress("UnsafeThirdPartyFunctionCall") // isMutable is always false
-            it.copy(it.config, false)
+            it.copy(Bitmap.Config.ARGB_8888, false)
```
Comment (0xnm) [inline features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: what is the reason for this change?

**Review by 0xnm — APPROVED** [review]

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt (hunk 2 above)
Comment (ambushwork) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: When testing with some other applications, I found that due to the original config of bitmap, some bitmaps are empty after copying it, for example `ALPHA_8`, I need to tweak the config in order to make it work.

Here is a related question on SO:
https://stackoverflow.com/questions/9247369/alpha-8-bitmaps-and-getpixel

Comment (0xnm) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: From SO:

> Simplest workaround is to waste lots of memory

Not sure though if it is useful to try to record `ALPHA_8` bitmaps, because documentation says:

```
 * Each pixel is stored as a single translucency (alpha) channel.
 * This is very useful to efficiently store masks for instance.
 * No color information is stored.
 * With this configuration, each pixel requires 1 byte of memory.
```

So probably it is a bitmask layer, rather than something that is rendered to the user. Also memory usage will be much higher - 4x more (pixel is 1 byte vs pixel is 4 bytes in cases of `ARGB_8888`).

Maybe we should keep here a TODO item for the further investigation? Maybe it makes sense to skip recording `ALPHA_8` bitmaps (if it won't affect anything).

Comment (ambushwork) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: theoretically yes it should be a translucency channel, but when I debug I see a bitmap like this:
image

and its original config is ALPHA_8. With the previous code it will copy nothing, but when changing to ARGB_888, it has real data.

Comment (0xnm) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: okay, makes sense then if we assume that with such config pixel is always just black and some alpha is applied. Maybe then we should use `ARGB_8888` only when original config is `ALPHA_8`? And use the original config otherwise.

**Review by xgouchet — APPROVED** [review]

File: features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt (hunk 2 above)
Comment (ambushwork) [reply features/dd-sdk-android-session-replay-compose/src/main/kotlin/com/datadog/android/sessionreplay/compose/internal/mappers/semantics/ImageSemanticsNodeMapper.kt:89]: I haven't test exhaustively all the config, but I will create a ticket to investigate all of them

File: features/dd-sdk-android-session-replay-compose/consumer-rules.pro (hunk 1 above)
Comment (ambushwork) [reply features/dd-sdk-android-session-replay-compose/consumer-rules.pro:52]: actually it depends on the usage, if in the code we want to check the class instance like this:
```
 if (ComposeReflection.WrappedCompositionClass?.isInstance(composition) == true) {

}
```
then only keeping the class members is not enough, we need to keep the class itself as well, otherwise we can just stick with `keepclassmembers`
